pkg/validator/container.go

```diff
@@ -182,6 +184,57 @@ func (cv *ContainerValidation) validateImage(imageConf *conf.Images) {
 			cv.addSuccess(messages.ImageTagSuccess, category, id)
 		}
 	}
+
+	// Check the container image against the user-defined image registry whitelist and blacklist.
+	imageWhiteLists := []*[]string{&imageConf.Whitelist.Error, &imageConf.Whitelist.Warning}
+	imageBlackLists := []*[]string{&imageConf.Blacklist.Error, &imageConf.Blacklist.Warning}
+	hasWhitelistCheck := false
+	hasWhitelistFailure := false
+	whitelistCheckID := "imageWhitelist"
+	hasBlacklistCheck := false
+	hasBlacklistFailure := false
+	blacklistCheckID := "imageBlacklist"
+
+	// Check whitelist
+	for _, imageList := range imageWhiteLists {
+		var severity config.Severity
+		if imageList == &imageConf.Whitelist.Error {
+			severity = config.SeverityError
+		} else {
+			severity = config.SeverityWarning
+		}
+		for _, imagePattern := range *imageList {
```

Comment:
Realizing now we didn't quite get this right. As it stands, the image has to pass _every_ whitelist check, so you couldn't specify e.g. `["docker.io/.*", "quay.io/.*"]`. Instead, we should fail only if _all_ the whitelist checks fail.

I think we just want to change `hasWhitelistFailure` to `hasWhitelistSuccess`. Then we can run
```go
for _, imagePattern := range *imageList {
  hasWhitelistCheck = true
  if regexp.MustCompile(imagePattern).MatchString(cv.Container.Image) {
    hasWhitelistSuccess = true
    break
  }
}
if hasWhitelistCheck {
  if hasWhitelistSuccess{
    addSuccess()
  } else {
    addFailure()
  }
}
```